lib/api/v3/work_packages/available_projects_on_create_api.rb

```diff
@@ -37,13 +37,26 @@ class AvailableProjectsOnCreateAPI < ::API::OpenProjectAPI
             authorize(:add_work_packages, global: true)
           end
 
+          params do
+            optional :for_type, type: Integer
+          end
+
           get do
             checked_permissions = Projects::ProjectCollectionRepresenter.checked_permissions
             current_user.preload_projects_allowed_to(checked_permissions)
 
             available_projects = WorkPackage
                                  .allowed_target_projects_on_create(current_user)
                                  .includes(Projects::ProjectCollectionRepresenter.to_eager_load)
+
+            if params[:for_type]
+              available_projects = available_projects.merge(
+                Project
+                  .joins(:types)
+                  .where("#{Type.table_name}.id = ?", params[:for_type])
+              )
+            end
```

Comment:
If we turn this into a proper project filter (like ancestor and active_or_archived):
* we can work on turning this into a standard end point using the end point helpers
* use a consistent interface for filtering in the API

Reply:
:+1: